raster-foundry/raster-foundry
# Add tags to campaigns; Add campaign clone endpoint

## Overview

This PR:
- adds a `tags` column to `campaigns` table
- adds an endpoint to enable cloning campaigns for users with `campaigns:clone` scope

### Checklist

- ~Swagger specification updated~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

## Testing Instructions

- CI should pass
- Add a new scope to the dev user in your local DB: `campaigns:clone`
- Run migrations
- Assemble api jar, spin up servers, start frontend, grab a `token`
- Clone the example campaign:
```bash
curl --location --request POST 'http://localhost:9091/api/campaigns/5c725cdf-e8af-4001-a3c4-98211d02f43e/clone' \
--header 'Authorization: Bearer <token>' \
--header 'Content-Type: application/json' \
--data-raw '{
	"tags": ["spring-2021"]
}'
```
- Make sure that the above works, and that the returned cloned campaign has the new tags, plus the `parentCampaignId` field is the original campaign ID

This is the backend part of issue 30 in that repo.
